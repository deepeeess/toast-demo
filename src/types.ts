export module Type {
  export interface Toast {
    text: string;
    timestamp: number;
    className?: string;
  }
}