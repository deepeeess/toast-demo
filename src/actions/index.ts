// import { SlideshowModel } from '../Models'
// import { History } from 'history';
// import * as fetch from 'isomorphic-fetch'


