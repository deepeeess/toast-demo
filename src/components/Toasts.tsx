import * as React from 'react';
import Toast from './Toast';
import { Type } from '../types';

interface ToastsProps {
  toasts: Type.Toast[]
  hideToast: (toast: Type.Toast) => any;
}

const Toasts = ({ toasts, hideToast }: ToastsProps) => {
  return (
    <div className="toast-container">
      { toasts.map((toast: Type.Toast, index) =>
        <Toast key={ index } toast={ toast } hideToast={ () => hideToast(toast) } />
      ) }
    </div>
  );
}

export default Toasts;