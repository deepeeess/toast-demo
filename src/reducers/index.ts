import {
  SHOW_TOAST,
  HIDE_TOAST,
  ToastAction
} from '../actions';
import { Type } from '../types';

export interface UiState {
  toasts: Type.Toast[];
}

const DEFAULT_STATE: UiState = {
  toasts: []
}

type ActionType = ToastAction | { type: '' };

export const uiReducers = (state: UiState = DEFAULT_STATE, action: ActionType): UiState => {
  switch (action.type) {
    case HIDE_TOAST:
      return {
        ...state,
        toasts: state.toasts.filter((toast: Type.Toast) => toast.timestamp !== action.toast.timestamp)
      }
    case SHOW_TOAST:
      return {
        ...state,
        toasts: [...state.toasts, action.toast]
      }
    default:
      return state;
  }
};

export default uiReducers;
