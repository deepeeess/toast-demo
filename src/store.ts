import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import uiReducers, { UiState } from './reducers';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({
  ui: uiReducers
});

export interface State {
  ui: UiState;
}

export const store = createStore(
  rootReducer,
  applyMiddleware(thunk)
);

store.subscribe(() => console.log(store.getState()));