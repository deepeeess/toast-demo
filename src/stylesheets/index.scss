@import 'colors';

$font-stack: "Palatino Linotype", "Book Antiqua", Palatino, serif;

body {
  position: absolute;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  font-family: $font-stack;
  color: #4C4845; 
}

.app-header {
  background-color: #222;
  height: 90px;
  padding-left: 20px;
  color: white;

  div {
    display: block;
    float: left;
  } 

  a {
    color: white;
  }
}

.app-title {
  text-align: left;

  h2 {
    padding-top: 2px;
    margin-bottom: 4px;
  }

  span {
    font-size: 10pt;
    vertical-align: top;
  }
}

.home-container {
  padding: 20px;
  height: 100%;
}

.button {
  color: white;
  font-family: $font-stack; 
  font-size: 12pt;
  margin: 8px;
  padding: 10px 16px;
}

.success {
  background-color: $teal;
}

.error {
  background-color: $red;
}

.toast {
  max-width: 600px;
  display: flex;
  flex-direction: row;
  color: white;
  padding: 8px;
  margin-bottom: 1px;

  .message {
    flex: 2 0 0;
  }

  .close {
    text-align: right;
  }
}
