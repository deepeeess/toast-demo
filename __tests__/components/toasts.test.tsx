import * as React from 'react';
import Toasts from '../../src/components/toasts';
import { shallow } from 'enzyme';
import { } from 'jest';
import toJson from 'enzyme-to-json'
import { Type } from '../../src/types';

describe('Toast', () => {

  it('renders', () => {
    const toasts: Type.Toast[] = [
      {
        text: "Test message",
        timestamp: 1497617358715
      },
      {
        text: "Message from another index",
        timestamp: 1497617358805
      }
    ];
    const wrapper = shallow(<Toasts toasts={ toasts } hideToast={ jest.fn() } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});