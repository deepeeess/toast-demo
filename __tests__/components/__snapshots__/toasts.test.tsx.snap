// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders 1`] = `
<div
  className="toast-container"
>
  <Toast
    hideToast={[Function]}
    toast={
      Object {
        "text": "Test message",
        "timestamp": 1497617358715,
      }
    }
  />
  <Toast
    hideToast={[Function]}
    toast={
      Object {
        "text": "Message from another index",
        "timestamp": 1497617358805,
      }
    }
  />
</div>
`;
